traversal_test.go

```diff
@@ -74,7 +74,7 @@ func TestReferenceTraversal(t *testing.T) {
 		elements int
 		refs     int
 	}{
-		{`{ "not" : { "$ref":"#" }}`, 2, 0},
+		{`{ "not" : { "$ref":"#" }}`, 3, 1},
```

Comment:
To me this test should expect 3 elements (`Schema`, `Not` and `Ref`), and 1 ref. The previous version was 2 and 0, but this feels wrong.